src/dsp_tools/analyse_xml_data/models.py

```diff
@@ -1,17 +1,24 @@
 from __future__ import annotations
 
+import uuid
 from dataclasses import dataclass, field
 
 
-@dataclass
+@dataclass(frozen=True)
 class ResptrLink:
     """This class represents a link between two resources."""
 
     subject_id: str
     object_id: str
+    link_uuid: str = field(default_factory=lambda: str(uuid.uuid4()))
+
+    @property
+    def cost_links(self) -> float:
+        """The cost of this outgoing is consistently 1"""
+        return 1
 
 
-@dataclass
+@dataclass(frozen=True)
 class XMLLink:
     """
     This class represents a link between a resource and an XML text
```

Comment:
This docstring is a bit cryptic:
 - actually, the link is not between the resource and the XML text
 - The 2nd line is grammatically out of place